src/pgduckdb_planner.cpp

```diff
@@ -50,21 +52,46 @@ PlanQuery(Query *parse, ParamListInfo bound_params) {
 	);
 }
 
-static Plan *
-CreatePlan(Query *query, const char *query_string, ParamListInfo bound_params) {
+std::tuple<duckdb::unique_ptr<duckdb::PreparedStatement>, duckdb::unique_ptr<duckdb::Connection>>
+DuckdbPrepare(Query *query, ParamListInfo bound_params) {
+	const char *query_string = pg_get_querydef(query, false);
 
-	List *rtables = query->rtable;
+	/* TODO: Move this state into custom_private probably */
```

Comment:
Putting an extra note here for visibility. This is very important to do, there can be big bugs otherwise. But I'd prefer to do this after merging #143, because that makes it easy to add tests for explain.

EXPLAIN in general is broken btw, if you use it on a extended protocol statement with parameters. That should be fixed too. The reason for that is that `Duckdb_ExplainCustomScan` doesn't provide the parameters to the duckdb `Execute` function yet. Basically we need to do there what I've now only done in `ExecuteQuery`, which is called by `Duckdb_ExecCustomScan`.

Reply:
EXPLAIN is now fixed. And I was wrong about it being necessary to move this to custom_private to avoid bugs. I added a comment in the `DuckdbExplainOneQueryHook` elaborating on why the current design is fine.